frontend/src/metabase/query_builder/components/view/NativeQueryPreviewSidebar/NativeQueryPreviewSidebar.tsx

```diff
@@ -36,10 +36,12 @@ export const NativeQueryPreviewSidebar = (): JSX.Element => {
   const sourceQuery = question.query();
   const canRun = Lib.canRun(sourceQuery);
   const payload = Lib.toLegacyQuery(sourceQuery);
-  const { data, error, isLoading } = useGetNativeDatasetQuery(payload);
+  const { data, error, isLoading, isFetching } =
+    useGetNativeDatasetQuery(payload);
 
-  const showError = canRun && error;
-  const showQuery = canRun && !error;
+  const showLoader = isLoading || isFetching;
```

Comment:
`isFetching` will be `true` if `isLoading` is `true`

Reply:
I was reading the source code and found this:
```js
/**
 * Query is currently loading for the first time. No data yet.
 */
isLoading: false;
/**
 * Query is currently fetching, but might have data from an earlier request.
 */
isFetching: false;
```

Unless I'm misinterpreting this, I think we want to cover both these cases. At first, when it's loading and then later when it's (re)fetching.

Do you understand it differently @kamilmielnik ?

Reply:
My point is that we don't need to use `isLoading` at all. `isFetching` covers all cases.

Reply:
Got it. Thanks.
I misunderstood this, and thought that `isFetching` will be false on the first load, which is not the case.